frontend/src/components/MainComponent/MainComponent.tsx

```diff
@@ -315,6 +319,38 @@ function MainComponent({
 		setChatModel({ ...chatModel, id: modelId });
 	}
 
+	// resets whole game progress and start from level 1 or Sandbox
+	async function resetProgress() {
+		console.log('resetting progress for all levels');
+		resetCompletedLevels();
+
+		const { emails, chatHistory, defences, chatModel } =
+			await resetService.resetAllLevelProgress(currentLevel);
+		processBackendLevelData(
```

Comment:
Is this processing necessary if we are on level 2 or 3 and we will be taken back to level 1?